libr/util/buf.c

```diff
@@ -652,8 +652,12 @@ R_API st64 r_buf_uleb128(RBuffer *b, ut64 *v) {
 			return -1;
 		}
 		c = data & 0xff;
-		sum |= ((ut64) (c & 0x7f) << s);
-		s += 7;
+		if (s < 64) {
+			sum |= ((ut64) (c & 0x7f) << s);
+			s += 7;
+		} else {
+			sum = 0;
```

Comment:
shall the function return as well? WDYT?

Reply:
or spit out a warning?, maybe we need to return error somehow as 3rd argument here so the caller can catch those issues properly?

Reply:
it seems right now errors are returned as -1. We could keep this behaviour maybe.

Reply:
-1 is fine for uleb, but not for sleb, imho it's conceptually wrong to mix data and error channels

Reply:
yeah :(